python-threatexchange/threatexchange/exchanges/impl/techagainstterrorism_api.py

```diff
@@ -0,0 +1,168 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+
+"""SignalExchangeAPI implementation for Tech Against Terrorism Hash List API"""
+
+
+from functools import lru_cache
+import json
+import tempfile
+import requests
+import time
+import typing as t
+from dataclasses import dataclass, field
+
+from threatexchange.exchanges.clients.techagainstterrorism import api
+
+from threatexchange.exchanges import fetch_state as state
+from threatexchange.exchanges import signal_exchange_api
+from threatexchange.exchanges import auth
+from threatexchange.exchanges.collab_config import (
+    CollaborationConfigBase,
+    CollaborationConfigWithDefaults,
+)
+from threatexchange.signal_type.signal_base import SignalType
+from threatexchange.signal_type.pdq.signal import PdqSignal
+from threatexchange.signal_type.md5 import VideoMD5Signal
+from threatexchange.signal_type.url import URLSignal
+from threatexchange.signal_type.raw_text import RawTextSignal
+
+
+@dataclass
+class TATCheckpoint(state.FetchCheckpointBase):
+    """
+    Tech Against Terrorism Hash List revolves around fetching
+    a JSON file which contains a list of hashes. The pre-signed URL
+    for the JSON file is returned from the API response.
+
+    The Hash List is updated nightly and new hashes are appended to the list.
+
+    For more information on our collection process please visit: https://terrorismanalytics.org/about/how-it-works
+    For our Hash List documentation: https://terrorismanalytics.org/docs/hash-list-v1
+    """
+
+    last_fetch_time: int
+
+    def is_stale(self) -> bool:
+        """
+        Consider stale after 24 hours of not fetching
+        """
+        return time.time() - self.last_fetch_time > 3600 * 24
+
+    @classmethod
+    def from_tat_fetch(cls, response: api.TATHashListResponse) -> "TATCheckpoint":
+        return cls(int(response.created_on.timestamp()))
+
+
+@dataclass
+class TATSignalMetadata(state.FetchedSignalMetadata):
+    """Not sure what should live here"""
+
+    pass
+
+
+@dataclass
+class TATCredentials(auth.CredentialHelper):
+    ENV_VARIABLE: t.ClassVar[str] = "TX_TAT_CREDENTIALS"
+    FILE_NAME: t.ClassVar[str] = "~/.tx_tat_credentials"
```

Comment:
I think the `TX_` in the threatexchange API refers to ThreatExchange the API. To avoid confusion, I'd use `PYTX_` as the full prefix or just leave it at `TAT_CREDENTIALS`